src/Nethermind/Nethermind.TxPool/Filters/RecoverAuthorityFilter.cs

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Crypto;
+
+namespace Nethermind.TxPool.Filters
+{
+    /// <summary>
+    /// Will recover authority from transactions with authority_list and filter any with bad signatures.
+    /// /// </summary>
+    internal sealed class RecoverAuthorityFilter(IEthereumEcdsa ecdsa) : IIncomingTxFilter
+    {
+        public AcceptTxResult Accept(Transaction tx, ref TxFilteringState state, TxHandlingOptions handlingOptions)
+        {
+            if (!tx.HasAuthorizationList)
+                return AcceptTxResult.Accepted;
+
+            Metrics.PendingTransactionsWithExpensiveFiltering++;
```

Comment:
This metric is incorrect here, as it counts transactions that reach the expensive filtering in TxPool. For SetCode transactions, we end up counting one transaction twice.

I think having a separate metric would be a good idea.

Reply:
There is already `Metrics.Eip7702TransactionsInBlock` so maybe this one should just be removed?

Reply:
removed